src/components/MultiStepIndicator/MultiStepIndicator.jsx

```diff
@@ -27,17 +27,22 @@ const MultiStepIndicator = forwardRef(
       onClick,
       textPlacement,
       id,
+      mode,
       "data-testid": dataTestId
     },
     ref
   ) => {
     const componentRef = useRef(null);
     const mergedRef = useMergeRefs({ refs: [ref, componentRef] });
+    const finalMode = useMemo(
+      () => (textPlacement === TEXT_PLACEMENTS.VERTICAL ? MODES.REGULAR : mode),
+      [mode, textPlacement]
+    );
 
     const renderHorizontalStepIndicator = useCallback(
       (step, index) => {
         return (
-          <>
+          <React.Fragment key={index}>
```

Comment:
why we need to add key here?

Reply:
Because it's mapped into array of step indicators below